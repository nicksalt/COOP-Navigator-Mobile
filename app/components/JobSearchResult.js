import React from "react";
import { View, StyleSheet, FlatList, Text } from "react-native";
import PropTypes from "prop-types";
import JobListItem from "./JobListItem";

export default class JobSearchResult extends React.Component {
  componentDidMount = () => {
    this._fetchJobs();
  };

  render() {
    if (this.props.errorOccurred) {
      return (
        <View>
          <Text>There was an error fetching jobs:</Text>
          <Text>{this.props.error.message}</Text>
        </View>
      );
    }

    return (
      <FlatList
        data={this.props.jobs}
        keyExtractor={job => job.jobId.toString()}
        refreshing={!!this.props.isFetching}
        onRefresh={this._fetchJobs}
        ItemSeparatorComponent={() => <View style={styles.separator} />}
        ListEmptyComponent={
          !this.props.isFetching && <Text>No matching jobs.</Text>
        }
        renderItem={({ item }) => (
          <JobListItem
            job={item}
            navigation={this.props.navigation}
            fetchSingleJobDispatch={this.props.fetchSingleJobDispatch}
          />
        )}
      />
    );
  }

  _fetchJobs = () => {
    this.props.fetchJobsDispatch();
  };
}

JobSearchResult.propTypes = {
  jobs: PropTypes.array,
  isFetching: PropTypes.bool,
  errorFetchingJobs: PropTypes.bool,
  error: PropTypes.object
};

const styles = StyleSheet.create({
  separator: {
    height: 1,
    backgroundColor: "#ccc"
  }
});
