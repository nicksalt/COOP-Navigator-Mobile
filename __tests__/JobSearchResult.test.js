import React from "react";
import JobSearchResult from "../app/components/JobSearchResult";
import renderer from "react-test-renderer";

jest.mock("FlatList", () => "FlatList");

test("JobSearchResult snapshot renders correctly", () => {
  let props = {};
  props.fetchJobsDispatch = () => {};
  props.isFetching = true;
  let tree = renderer.create(<JobSearchResult {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
  props.isFetching = false;
  props.errorOccurred = true;
  props.error = { message: "ERROR OCCURRED" };
  tree = renderer.create(<JobSearchResult {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
  props.errorOccurred = false;
  props.jobs = [];
  tree = renderer.create(<JobSearchResult {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
