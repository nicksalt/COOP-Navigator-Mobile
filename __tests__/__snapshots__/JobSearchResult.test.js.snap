// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobSearchResult snapshot renders correctly 1`] = `
<FlatList
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={false}
  keyExtractor={[Function]}
  onRefresh={[Function]}
  refreshing={true}
  renderItem={[Function]}
/>
`;

exports[`JobSearchResult snapshot renders correctly 2`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    There was an error fetching jobs:
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    ERROR OCCURRED
  </Text>
</View>
`;

exports[`JobSearchResult snapshot renders correctly 3`] = `
<FlatList
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      No matching jobs.
    </Text>
  }
  data={Array []}
  keyExtractor={[Function]}
  onRefresh={[Function]}
  refreshing={false}
  renderItem={[Function]}
/>
`;
